src/index.js

```diff
@@ -15,28 +15,45 @@ const CHUNK_SIZE = 262144
 // Use a layout + chunkers to convert a directory (or file) to the layout format
 exports.import = (options, callback) => {
   // options.path : what to import
+  // options.buffer : import a buffer
+  // options.filename : optional file name for buffer 
+  // options.stream : import a stream
   // options.recursive : follow dirs
   // options.chunkers : obj with chunkers to each type of data, { default: dumb-chunker }
   // options.dag-service : instance of block service
   const dagService = options.dagService
 
-  const stats = fs.statSync(options.path)
-  if (stats.isFile()) {
-    fileImporter(options.path, callback)
-  } else if (stats.isDirectory() && options.recursive) {
-    dirImporter(options.path, callback)
-  } else {
-    return callback(new Error('recursive must be true to add a directory'))
+  if(options.buffer) {
+    if(!Buffer.isBuffer(options.buffer)) { 
+      return callback(new Error('buffer importer must take a buffer'))
+    }
+    bufferImporter(options.buffer, callback)
+  } else if(options.stream) {
+    if(!(typeof options.stream.on === 'function')) {
+      return callback(new Error('stream importer must take a readable stream'))
+    }
+    streamImporter(options.stream, callback)
+  } 
```

Comment:
there should be a `return` statement here somewhere, so that if someone passes a path and a buffer, it doesn't fire two importers